### PR TITLE
Refactor duckhunt .killers/.friends commands

### PR DESCRIPTION
**Pull request by linuxdaemon**
Move all common code to a central function

Fix issues in certain error responses

### CODE REVIEW

File: plugins/duckhunt.py [hunk 1]
```diff
@@ -2,18 +2,29 @@
 import random
 from collections import defaultdict
 from threading import Lock
-from time import time, sleep
+from time import sleep, time
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py]: ....why?

File: plugins/duckhunt.py [hunk 2]
```diff
@@ -2,18 +2,29 @@
...
 
 from cloudbot import hook
 from cloudbot.event import EventType
 from cloudbot.util import database
 from cloudbot.util.formatting import pluralize_auto, truncate
+from cloudbot.util.func_utils import call_with_args
 
 duck_tail = "・゜゜・。。・゜゜"
-duck = ["\\_o< ", "\\_O< ", "\\_0< ", "\\_\u00f6< ", "\\_\u00f8< ", "\\_\u00f3< "]
+duck = [
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py]: Own PR for cleanup

File: plugins/duckhunt.py [hunk 3]
```diff
@@ -70,10 +81,10 @@ def should_deploy(self, conn):
         msg_delay = get_config(conn, 'minimum_messages', 10)
         mask_req = get_config(conn, 'minimum_users', 5)
         return (
-            self.game_on and self.duck_status == 0 and
-            self.next_duck_time <= time() and
-            self.messages >= msg_delay and
-            len(self.masks) >= mask_req
+                self.game_on and self.duck_status == 0 and
```
Comment (A-UNDERSCORE-D) [inline plugins/duckhunt.py]: Own PR for cleanup

File: tests/plugin_tests/test_duckhunt.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    duckhunt.update_score('TestUser', '#TestChannel', session, conn, 5, 4)
+    duckhunt.update_score('TestUser1', '#TestChannel', session, conn, 1, 7)
+    duckhunt.update_score('OtherUser', '#TestChannel1', session, conn, 9, 2)
+
+    assert duckhunt.get_channel_scores(
+        session, duckhunt.SCORE_TYPES['friend'], conn, '#TestChannel'
+    ) == {'testuser': 4, 'testuser1': 7}
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    class Conn:
+        name = 'TestConn'
+
+    conn = Conn()
+
+    duckhunt.update_score('TestUser', '#TestChannel', session, conn, 5, 4)
+    duckhunt.update_score('TestUser1', '#TestChannel', session, conn, 1, 7)
+    duckhunt.update_score('OtherUser', '#TestChannel1', session, conn, 9, 2)
+
+    assert duckhunt.get_channel_scores(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    ) == {'testuser': 4, 'testuser1': 7}
+
+    assert repr(
+        duckhunt.friends('', '#TestChannel', conn, session)
+    ) == repr(
+        'Duck friend scores in #TestChannel: '
+        '\x02t\u200bestuser1\x02: 7 • \x02t\u200bestuser\x02: 4'
+    )
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        duckhunt.killers('', '#TestChannel', conn, session)
+    ) == repr(
+        'Duck killer scores in #TestChannel: '
+        '\x02t\u200bestuser\x02: 5 • \x02t\u200bestuser1\x02: 1'
+    )
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        duckhunt.friends('global', '#TestChannel', conn, session)
+    ) == repr(
+        'Duck friend scores across the network: '
+        '\x02t\u200bestuser1\x02: 7'
+        ' • \x02t\u200bestuser\x02: 4'
+        ' • \x02o\u200btheruser\x02: 2'
+    )
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        duckhunt.killers('global', '#TestChannel', conn, session)
+    ) == repr(
+        'Duck killer scores across the network: '
+        '\x02o\u200btheruser\x02: 9'
+        ' • \x02t\u200bestuser\x02: 5'
+        ' • \x02t\u200bestuser1\x02: 1'
+    )
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        duckhunt.friends('average', '#TestChannel', conn, session)
+    ) == repr(
+        'Duck friend scores across the network: '
+        '\x02t\u200bestuser1\x02: 7'
+        ' • \x02t\u200bestuser\x02: 4'
+        ' • \x02o\u200btheruser\x02: 2'
+    )
+
+    assert repr(
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Use variables instead. This is a mess to read

File: tests/plugin_tests/test_duckhunt.py [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+
+@pytest.fixture()
+def mock_db():
+    return MockDB()
+
+
+@pytest.mark.parametrize('prefix,items,result', [
+    (
+            'Duck friend scores in #TestChannel: ',
```
Comment (A-UNDERSCORE-D) [inline tests/plugin_tests/test_duckhunt.py]: Indentation looks weird here

**Review by A-UNDERSCORE-D — CHANGES_REQUESTED** [review]

File: plugins/duckhunt.py (hunk 1 above)
Comment (linuxdaemon) [reply plugins/duckhunt.py]: pycharm import organization

**Review by A-UNDERSCORE-D — APPROVED** [review]

**Pull request merged**